src/minimal_unified_api.py

```
@@ -0,0 +1,250 @@
+        raise HTTPException(status_code=400, detail=str(e)) from e
+    except Exception as e:
+        logger.exception("Error in emotion analysis")
+        raise HTTPException(status_code=500, detail="Emotion analysis failed") from e
+
+@app.post("/analyze/summarize")
+async def summarize_text(text: str):
+    """Summarize text using T5 model."""
+    try:
+        # Input validation
+        if not text or not isinstance(text, str) or not text.strip():
+            raise HTTPException(status_code=400, detail="Text input is required and cannot be empty")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)